src/main/java/org/fsg1/fmms/backend/database/Connection.java

```diff
@@ -43,18 +43,107 @@ public Connection(final Configuration config, final BasicDataSource connectionPo
      * @throws Exception if something goes wrong performing the query.
      */
     public String executeQuery(final String columnName, final String query, final Object... parameters) throws Exception {
-        try (java.sql.Connection connection = connectionPool.getConnection()) {
-            try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
-                mapParams(preparedStatement, parameters);
+        try (java.sql.Connection connection = startTransaction()) {
+            final String result = executeQuery(connection, columnName, query, parameters);
+            commitTransaction(connection);
+            return result;
+        }
+    }
+
+    /**
+     * Offers you a Connection from the connection pool and disables auto committing. Any queries executed on
+     * this connection will not go in effect until the commitTransaction method is called.
+     * Keep in mind the timeouts will be the driver's defaults.
+     * This connection will not be closed until the transaction is committed with the commitTransaction method.
+     *
+     * @return A connection to do transactions with.
+     * @throws SQLException If a database access error occurs, while participating in a distributed transaction
+     *                      or this method is called on a closed connection.
+     */
+    public java.sql.Connection startTransaction() throws SQLException {
+        final java.sql.Connection connection = connectionPool.getConnection();
+        connection.setAutoCommit(false);
+        return connection;
+    }
 
-                try (ResultSet result = preparedStatement.executeQuery()) {
-                    if (!result.next() || result.getString(columnName) == null) throw new EntityNotFoundException();
-                    return result.getString(columnName);
-                }
+    /**
+     * Executes a statement on the given connection. This statement will be executed but not committed as it is
+     * in an open transaction until the commitTransaction method is called.
+     * If an exception occurs at any time during the transaction it is rollbacked and aborted.
+     *
+     * @param connection Connection to execute the statement on.
+     * @param columnName Column name to retrieve any possible result. If no column name is given, the statement is seen
+     *                   as an update instead of a query.
+     * @param statement  Statement to perform.
+     * @param parameters Array of parameters to map to the statement.
+     * @return A possible result in JSON string form if columnName is given, else an empty string is returned.
+     * @throws SQLException If a database access error occurs or anything else goes wrong.
+     */
+    public String executeQuery(final java.sql.Connection connection,
```

Comment:
There is no need to run select queries in a transaction. So acquire your own connection an close it. No external dependencies